include/kernel.h

```diff
@@ -123,6 +123,103 @@ struct k_timer;
 struct k_poll_event;
 struct k_poll_signal;
 
+enum k_objects {
+	K_OBJ_ALERT,
+	K_OBJ_DELAYED_WORK,
+	K_OBJ_MEM_SLAB,
```

Comment:
Are slabs really needed?  Seems like (just like pools) the API should work just fine from user code when referring to memory inside userspace, and the utility of managing a kernel-side allocator from user code that can't touch it seems kinda weird.

Reply:
in its current implementation, the mem_slab APIs will not work from userspace. they contain wait queues, they have other threads pending on them, etc...redesign would be needed to make them work entirely from userspace. have a look at the code for k_mem_slab_free() there's all kinds of stuff in there that userspace can't do.

also its worth noting that detailed analysis of kernel object APIs is for a later stage in development of this feature, currently I am not changing any of them and just trying to get the infrastructure in place.